compact.go

```diff
@@ -471,13 +481,19 @@ func (w *instrumentedChunkWriter) WriteChunks(chunks ...chunks.Meta) error {
 	return w.ChunkWriter.WriteChunks(chunks...)
 }
 
+// ErrCompactionCanceled is returned when the compaction was canceled during shutdown.
+var ErrCompactionCanceled = errors.New("compaction cancelled")
+
 // write creates a new block that is the union of the provided blocks into dir.
 // It cleans up all files of the old blocks after completing successfully.
 func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockReader) (err error) {
 	dir := filepath.Join(dest, meta.ULID.String())
 	tmp := dir + ".tmp"
-
+	var writers []io.Closer
 	defer func(t time.Time) {
+		for _, w := range writers {
+			w.Close()
```

Comment:
Please do not ignore errors, especially on this path. 

We need something more complex like we have in Thanos:  https://github.com/improbable-eng/thanos/blob/master/pkg/runutil/runutil.go#L112

(Not only log line, but actually adding some multierror to `err` return variable, even in defer.

This is important because write might be not successful if index close fails etc...

Reply:
And I know it is no regression, but if we are fixing this let's fix it right. Not saying we need helper for this like Thanos have, but helper would actually makes the code bit simpler (:

Reply:
yes this was bugging me as well and I think I managed to implement some nice err handling.